app/services/earnings_csv_parser.rb

```diff
@@ -0,0 +1,83 @@
+require 'csv'
+require_relative '../models/earning'  # Add this line if Earning model is not being loaded
+
+class EarningsCsvParser
+  STRING = 'STRING'.freeze
+  DECIMAL = 'DECIMAL'.freeze
+  # file = CSV string
+  def initialize(file, employer_id)
+      @file = file
+      @employer = get_employer(employer_id)
+      @employer_csv_format = get_employer_csv_format
+  end
+
+  def create_earnings
+      csv = CSV.parse(@file, headers: true)
+      errors = []
+      csv.each do |row|
+        e = create_earning(row)
+        if e.valid?
+          e.save
+        else
+          errors << "Error with row #{row}: #{e.errors.full_messages}"
+        end
+      end
+      display_errors errors
+  end
+
+  private
+
+  def get_employer_csv_format
+    EmployerCsvFormat.find_by!(employer_id: @employer.id)
+  end
+  
+  def get_employer employer_id
+    Employer.find_by!(id: employer_id) 
```

Comment:
`find_by!` and passing in the primary key is the same as calling `find(<id>)` (which will raise an exception if the record is not found)

```suggestion
    Employer.find(employer_id) 
```